### ISSUE TITLE
[Bug] Work stealing doesn't work

### ISSUE THREAD

**Issue by lixin-wei** [actor-framework/actor-framework#1866]
Reproduce code:
```cpp
#include "caf/actor_system.hpp"
#include "caf/behavior.hpp"
#include "caf/event_based_actor.hpp"
#include "caf/exec_main.hpp"
#include "caf/timespan.hpp"

#include <chrono>
#include <thread>

using namespace caf;
using namespace std::literals;

size_t GetThreadID() {
  return std::hash<std::thread::id>{}(std::this_thread::get_id());
}

behavior actor_want_to_be_stolen(event_based_actor* self) {
  return {[self](int _) {
    self->println("Got msg, callee thread_id={}", GetThreadID());
  }};
}

void heavy_actor(event_based_actor* self, const actor& buddy) {
  // If you can't reproduce, try increase loop num
  for (int i = 0; i < 10; ++i) {
    self->println("Send msg, caller thread_id={}", GetThreadID());
    self->mail(1).request(buddy, infinite);
    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
  }
}

void caf_main(actor_system& sys) {
  auto actor = sys.spawn(actor_want_to_be_stolen);
  sys.spawn(heavy_actor, actor);
}

CAF_MAIN()

```

This is a very simple code, `heavy_actor` is a actor method which will take long time(simulated by using `sleep`), while `heavy_actor` is doing his work, I expect `actor_want_to_be_stolen` can be run in other threads.

However, the code outputs:
```
-> % ./test
Send msg, caller thread_id=6060986316492724180
Got msg, callee thread_id=8012040026766645660
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Send msg, caller thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
Got msg, callee thread_id=6060986316492724180
```

`Got msg` only show few times(mostly once) and never show again, until `heavy_actor` has done his work.

Which means `actor_want_to_be_stolen` is executed in other threads few times. **But once it is scheduled to the same thread as `heavy_actor`, it never run again, also won't be stolen by other threads.**

Could anyone help me? Thanks in advance!

**Comment by lixin-wei** [comment]
@Neverlord Thanks for the reply, may I ask an expected date of the fixing?

If you don't have time, would you please say something about the root cause? I'm very willing to help.😀

**Comment by Neverlord** [comment]
Looks like this regression got introduced in a 0.19 release, I have a PR ready to fix the issue. The patch will be included in 1.0 (which we are currently finalizing).

**Comment by lixin-wei** [comment]
Thank you so much for the quick fix! ❤️
